spikeinterface/toolkit/preprocessing/deepinterpolation.py

```diff
@@ -0,0 +1,195 @@
+from .basepreprocessor import BasePreprocessor, BasePreprocessorSegment
+import numpy as np
+
+try:
+    from tensorflow.keras.models import load_model
+    HAVE_TF = True
+except ImportError:
+    HAVE_TF = False
+    
+class DeepInterpolatedRecording(BasePreprocessor):
+    name = 'deepinterpolate'
+    
+    def __init__(self, recording, path_to_model='./2020_02_29_15_28_unet_single_ephys_1024_mean_squared_error-1050.h5', 
+                 pre_frames=30, post_frames=30, pre_post_omission=1, 
+                 n_frames_normalize=20000):
+        
+        assert HAVE_TF, "To use deep interpolation, you need to install tensorflow first."
+        
+        BasePreprocessor.__init__(self, recording)
+        
+        # load model
+        self.model = load_model(filepath=path_to_model)
+        
+        # estimate mean and std
+        if n_frames_normalize > recording.get_num_frames():
+            print('Too few frames; using all frames to estimate mean and std')
+            n_frames_normalize = recording.get_num_frames()
+            
+        local_data = recording.get_traces(start_frame=0, 
+                                          end_frame=n_frames_normalize)
+
+        local_mean = np.mean(local_data.flatten())
+        local_std = np.std(local_data.flatten())
```

Comment:
Maybe MAD would be more robust no ?

Reply:
yeah we can try it, it's just that the original implementation by Jerome uses std